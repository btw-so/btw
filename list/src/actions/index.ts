export * from './alerts';
export * from './app';
export * from './github';
export * from './user';
export * from './list';
export * from './notes';