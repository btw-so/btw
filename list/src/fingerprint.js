import md5 from './md5';

var SEP = '|';
var ua = window.navigator.userAgent.toLowerCase();
var opera = ua.indexOf('opera') >= 0 || ua.indexOf('opr') >= 0;
var ie = ua.indexOf('msie') >= 0 && !opera;

/**
 *
 */
function activeXDetect(componentClassID) {
  var e,
    obj = document.createElement('div'),
    verIEfull = null; // Full IE version [string/null]

  try {
    obj.style.behavior = 'url(#default#clientcaps)';
    verIEfull = obj
      .getComponentVersion('{' + componentClassID + '}', 'componentid')
      .replace(/,/g, '.');
  } catch (e) {}

  return verIEfull;
}

/**
 * Remove illegal characters from a string
 *
 * @return string
 */
function stripIllegalChars(value) {
  return value.replace(/[\n\/\\]/g, '');
}

/**
 * Get filename from a full path string.
 *
 * @return string
 */
function baseName(tempFileName) {
  return tempFileName.replace(/.*[\/\\]/g, '');
}

/**
 * Get the plugins information.
 *
 * @return string
 */
function fingerprint_software() {
  var t = '';
  var isFirst = true;

  if (window.navigator.plugins.length > 0) {
    for (var i = 0; i < window.navigator.plugins.length; i++) {
      var plugin = window.navigator.plugins[i];
      if (isFirst == true) {
        t += baseName(plugin.filename);
        isFirst = false;
      } else {
        t += SEP + baseName(plugin.filename);
      }
    }
  } else if (window.navigator.mimeTypes.length > 0) {
    for (var i = 0; i < window.navigator.mimeTypes.length; i++) {
      var mimeType = window.navigator.mimeTypes[i];
      if (isFirst == true) {
        t += mimeType.type;
        isFirst = false;
      } else {
        t += SEP + mimeType.type;
      }
    }
  } else if (ie) {
    var components = new Array(
      '7790769C-0471-11D2-AF11-00C04FA35D02',
      '89820200-ECBD-11CF-8B85-00AA005B4340',
      '283807B5-2C60-11D0-A31D-00AA00B92C03',
      '4F216970-C90C-11D1-B5C7-0000F8051515',
      '44BBA848-CC51-11CF-AAFA-00AA00B6015C',
      '9381D8F2-0288-11D0-9501-00AA00B911A5',
      '4F216970-C90C-11D1-B5C7-0000F8051515',
      '5A8D6EE0-3E18-11D0-821E-444553540000',
      '89820200-ECBD-11CF-8B85-00AA005B4383',
      '08B0E5C0-4FCB-11CF-AAA5-00401C608555',
      '45EA75A0-A269-11D1-B5BF-0000F8051515',
      'DE5AED00-A4BF-11D1-9948-00C04F98BBC9',
      '22D6F312-B0F6-11D0-94AB-0080C74C7E95',
      '44BBA842-CC51-11CF-AAFA-00AA00B6015B',
      '3AF36230-A269-11D1-B5BF-0000F8051515',
      '44BBA840-CC51-11CF-AAFA-00AA00B6015C',
      'CC2A9BA0-3BDD-11D0-821E-444553540000',
      '08B0E5C0-4FCB-11CF-AAA5-00401C608500',
      'D27CDB6E-AE6D-11CF-96B8-444553540000',
      '2A202491-F00D-11CF-87CC-0020AFEECF20',
    );
    for (var i = 0; i < components.length; i++) {
      ver = activeXDetect(components[i]);

      if (ver !== null) {
        if (isFirst == true) {
          t += ver;
          isFirst = false;
        } else {
          t += SEP + ver;
        }
      } else {
        t += SEP + 'null';
      }
    }
  }

  return t;
}

/**
 * Get the final browser fingerprint (as hash, eventually).
 *
 * @params boolean hash Flag to make a message digest of the fingerprint
 *
 * @return string
 */
function get_fingerprint(hash) {
  var data = new Array(
    window.navigator.userAgent.toLowerCase(), // User Agent Information
    [screen.width, screen.height, screen.colorDepth].join(SEP), // Screen Information
    new Date().getTimezoneOffset(), // Time Zone Information
    !!window.sessionStorage, // Session Storage Information
    !!window.localStorage, // Local Storage Information
    window.navigator.platform, // Operative System Information
    navigator.cookieEnabled, // Cookie Enabled Information
    navigator.javaEnabled(), // Java Enabled Information
    navigator.language, // Language Information
    fingerprint_software(), // Plugins Information
  );

  var fingerprint = data.join('');
  if (hash) {
    fingerprint = '';
    for (var idx in data) {
      fingerprint += md5(data[idx].toString().toLowerCase());
    }
  }

  return fingerprint;
}

function get_map(fingerprint) {
  return fingerprint
    .replace(/[0-9]/g, '.')
    .replace(/[a-f]/g, '+')
    .replace(/\.\./g, ':')
    .replace(/\+\+/g, '#')
    .replace(/##/g, '@')
    .replace(/::/g, '~');
}

// function final_fingerprint() {
//   var fingerprint = get_fingerprint(true);
//   var fingerprint_hash = md5(fingerprint);

//   var map = get_map(fingerprint);
//   var map_hash = md5(map);
//   return fingerprint_hash + map_hash;
// }

function final_fingerprint() {
  var key = 'fingerprint_uuid';
  // check if there is a uuid in local storage, then return it
  // create a random uuid and save it in local storage
  // return the uuid
  var uuid = localStorage.getItem(key);
  if (uuid) {
    return uuid;
  }
  uuid = Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);

  localStorage.setItem(key, uuid);
  return uuid;
}

export default final_fingerprint;
