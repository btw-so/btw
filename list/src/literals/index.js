import { keyMirror } from "@gilbarbara/helpers";

// import { Status } from "types";

export const ActionTypes = keyMirror({
  GITHUB_GET_REPOS_REQUEST: undefined,
  GITHUB_GET_REPOS_SUCCESS: undefined,
  GITHUB_GET_REPOS_FAILURE: undefined,
  HIDE_ALERT: undefined,
  SHOW_ALERT: undefined,
  SET_APP_OPTIONS: undefined,
  USER_LOGIN_REQUEST: undefined,
  USER_LOGIN_SUCCESS: undefined,
  USER_LOGIN_FAILURE: undefined,
  USER_LOGOUT_REQUEST: undefined,
  USER_LOGOUT_SUCCESS: undefined,
  USER_LOGOUT_FAILURE: undefined,
  GET_USER: undefined,
  GET_USER_SUCCESS: undefined,
  GET_USER_FAILURE: undefined,
  GENERATE_OTP: undefined,
  GENERATE_OTP_SUCCESS: undefined,
  GENERATE_OTP_FAILURE: undefined,
  VERIFY_OTP: undefined,
  VERIFY_OTP_SUCCESS: undefined,
  VERIFY_OTP_FAILURE: undefined,
  UPDATE_USER: undefined,
  UPDATE_USER_SUCCESS: undefined,
  UPDATE_USER_FAILURE: undefined,
  RESET_USER_STATE: undefined,
  GET_LIST: undefined,
  GET_LIST_SUCCESS: undefined,
  GET_LIST_FAILURE: undefined,
  NEW_LIST_NODE: undefined,
  EDIT_LIST_NODE_POS: undefined,
  EDIT_LIST_NODE_CONTENT: undefined,
  UPSERT_LIST_NODE: undefined,
  CHANGE_SELECTED_NODE: undefined,
  BATCH_PUSH_NODES: undefined,
  SAVE_NOTE_CONTENT: undefined,
  GET_NOTE: undefined,
  GET_NOTE_SUCCESS: undefined,
  GET_NOTE_FAILURE: undefined,
  ADD_FILE: undefined,
  ADD_FILE_SUCCESS: undefined,
  ADD_FILE_FAILURE: undefined,
  GET_FILE: undefined,
  GET_FILE_SUCCESS: undefined,
  GET_FILE_FAILURE: undefined,
  GET_PINNED_NODES: undefined,
  GET_PINNED_NODES_SUCCESS: undefined,
  GET_PINNED_NODES_FAILURE: undefined,
  GET_PUBLIC_NOTE: undefined,
  GET_PUBLIC_NOTE_SUCCESS: undefined,
  GET_PUBLIC_NOTE_FAILURE: undefined,
  SEARCH_NODES: undefined,
  SEARCH_NODES_SUCCESS: undefined,
  SEARCH_NODES_FAILURE: undefined,
});

export const STATUS = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error",
  PINNED: "pinned",
};
