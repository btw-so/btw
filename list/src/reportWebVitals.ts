import { ReportHandler } from 'web-vitals';

const reportWebVitals = (onPerfEntry?: ReportHandler) => {
  if (onPerfEntry && onPerfEntry instanceof Function) {
    import('web-vitals').then(({ getCLS, getFCP, getFID, getLCP, getTTFB }) => {
      getCLS(onPerfEntry);
      getFID(onPerfEntry);
      getFCP(onPerfEntry);
      getLCP(onPerfEntry);
      getTTFB(onPerfEntry);
    });
  }
};

export default reportWebVitals;
