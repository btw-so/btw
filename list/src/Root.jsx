import React, { useEffect } from "react";
import { Helmet } from "react-helmet-async";
import { useDispatch } from "react-redux";
import {
  BrowserRouter,
  HashRouter,
  Route,
  Routes,
  useNavigate,
} from "react-router-dom";
import { selectUser } from "selectors";
import useTreeChanges from "tree-changes-hook";
import { useAppSelector } from "modules/hooks";

import toast, { Toaster } from "react-hot-toast";

import { getUser } from "actions";
// import Home from 'routes/Home';
import Login from "./routes/Login";
import NotFound from "routes/NotFound";
import PublicRoute from "./components/PublicRoute";
import PrivateRoute from "./components/PrivateRoute";
import Settings from "./routes/Settings";
import List from "./routes/List";
import PublicNote from "./routes/PublicNote";

function Root() {
  const dispatch = useDispatch();
  const userState = useAppSelector(selectUser);
  const { changed } = useTreeChanges(userState);

  const { user } = userState;
  const { isLoggedIn } = user;

  const userDomain =
    ((user.data || {}).domains || []).length > 0
      ? user.data.domains[0].domain
      : (user.data || {}).slug
      ? `${(user.data || {}).slug}.btw.so`
      : null;

  useEffect(() => {
    // get user details if it is the first time
    dispatch(getUser());
  }, []);

  useEffect(() => {
    if (changed("user.isLoggedIn", true) && isLoggedIn) {
      toast.success("Logged In!");
    }
  }, [dispatch, changed]);

  // THOUGHTS 101
  // create a component that checks if the environment is electron, then it would be BrowserRouter
  // otherwise, it would be HashRouter
  // this is because electron does not support hash routing
  // but the web version does
  // so, we need to check if it is electron or not
  // and then use the correct router
  // this is a temporary solution
  // we need to find a better way to do this
  // maybe we can use a custom hook to check if it is electron or not
  // and then use the correct router

  // THOUGHTS 102
  // For some reason, hard refresh on the urls is giving 404 if we use BrowserRouter in production

  // const Router = process.env.REACT_APP_ELECTRON ? HashRouter : BrowserRouter;
  // const Router = HashRouter;
  const Router = BrowserRouter;

  Helmet.defaultProps.encodeSpecialCharacters = false;

  return (
    <Router>
      <div className="w-full h-full flex-grow flex flex-col">
        <Helmet
          defaultTitle={"btw ∴"}
          defer={false}
          encodeSpecialCharacters={false}
          titleAttributes={{ itemprop: "name", lang: "en-en" }}
          titleTemplate={`%s | btw ∴`}
        >
          <link
            href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,700;1,400;1,700&display=swap"
            rel="stylesheet"
          />
        </Helmet>

        {/* {isLoggedIn && <Login />} */}
        {/* <Main isAuthenticated={isAuthenticated}> */}
        <Routes className="flex flex-grow">
          <Route
            className="flex flex-grow"
            element={
              <PrivateRoute isLoggedIn={isLoggedIn} to="/login">
                <List
                  userId={user && user.data ? user.data.id : null}
                  name={user && user.data ? user.data.name : null}
                  email={user && user.data ? user.data.email : null}
                  adminUser={
                    user &&
                    user.data &&
                    user.data.pro &&
                    user.data.email &&
                    [
                      "siddhartha.gunti191@gmail.com",
                      "deepti.vchopra@gmail.com",
                    ].includes(user.data.email)
                  }
                />
              </PrivateRoute>
            }
            path="/list"
          />
          <Route
            className="flex flex-grow"
            element={
              <PrivateRoute isLoggedIn={isLoggedIn} to="/login">
                <Settings
                  userId={user && user.data ? user.data.id : null}
                  name={user && user.data ? user.data.name : null}
                  email={user && user.data ? user.data.email : null}
                  share_id={
                    (user &&
                    user.data &&
                    user.data.domains &&
                    user.data.domains.length > 0
                      ? user.data.domains[0].share_id
                      : null) ||
                    (user && user.data && user.data.share_id
                      ? user.data.share_id
                      : null)
                  }
                  adminUser={
                    user &&
                    user.data &&
                    user.data.pro &&
                    user.data.email &&
                    [
                      "siddhartha.gunti191@gmail.com",
                      "deepti.vchopra@gmail.com",
                    ].includes(user.data.email)
                  }
                />
              </PrivateRoute>
            }
            path="/settings"
          />
          <Route
            element={
              <PublicRoute isLoggedIn={isLoggedIn} to="/list">
                <Login />
              </PublicRoute>
            }
            path="/login"
          />
          <Route
            element={
              <PublicRoute isLoggedIn={isLoggedIn} to="/list">
                <Login />
              </PublicRoute>
            }
            path="/"
          />
          <Route
            path="/public/note/:id/:hash"
            element={<PublicNote />}
          />
          <Route element={<NotFound />} path="*" />
        </Routes>
        <Toaster />
        {/* </Main> */}
        {/* <Footer /> */}
      </div>
    </Router>
  );
}

export default Root;
