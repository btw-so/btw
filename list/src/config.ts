import { STATUS } from 'literals';

import { Topic } from 'types';

export const description = 'Boilerplate with React and Redux with Redux Saga';
export const name = 'React Redux Saga Boilerplate';
export const topic: Topic = {
  cached: false,
  data: [],
  message: '',
  status: STATUS.IDLE,
  updatedAt: 0,
};
