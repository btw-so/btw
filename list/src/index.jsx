import React from "react";
import ReactDOM from "react-dom";
import { ErrorBoundary } from "react-error-boundary";
import { HelmetProvider } from "react-helmet-async";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/lib/integration/react";
import { configStore } from "store";

import { showAlert } from "actions";

import ErrorHandler from "components/ErrorHandler";
import Loader from "components/Loader";
import Reload from "components/Reload";
import GlobalStyles from "containers/GlobalStyles";

import reportWebVitals from "./reportWebVitals";
import Root from "./Root";
import { register } from "./serviceWorkerRegistration";

const { persistor, store } = configStore();

window.store = store;

ReactDOM.render(
  <Provider store={store}>
    <PersistGate loading={<Loader block size={100} />} persistor={persistor}>
      <ErrorBoundary FallbackComponent={ErrorHandler}>
        <HelmetProvider>
          <Root />
        </HelmetProvider>
      </ErrorBoundary>
      <GlobalStyles />
    </PersistGate>
  </Provider>,
  document.getElementById("root")
);

/* istanbul ignore next */
// register({
//   onUpdate: () => {
//     store.dispatch(showAlert(<Reload />, { id: 'sw-update', icon: 'bolt', timeout: 0 }));
//   },
// });

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals(console.log); // eslint-disable-line no-console
