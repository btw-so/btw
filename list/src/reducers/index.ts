import alerts, { alertsState } from './alerts';
import app, { appState } from './app';
import github, { githubState } from './github';
import user, { userState } from './user';
import otp, { otpState } from './user';
import list, { listState } from './list';

export const initialState = {
  alerts: alertsState,
  app: appState,
  github: githubState,
  user: userState,
  otp: otpState,
  list: listState
};

export default {
  ...alerts,
  ...app,
  ...github,
  ...user,
  ...otp,
  ...list
};
