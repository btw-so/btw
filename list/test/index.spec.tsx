import React from 'react';

jest.mock('redux-persist/lib/integration/react', () => ({
  PersistGate({ children }) {
    return <div>{children}</div>;
  },
}));

jest.mock('react-helmet-async', () => ({
  HelmetProvider({ children }) {
    return children;
  },
  Helmet() {
    return null;
  },
}));

jest.mock('../src/Root', () => () => <div>root</div>);

describe('src/index', () => {
  beforeAll(() => {
    process.env.NODE_ENV = 'production';
    const root = document.createElement('div');

    root.id = 'root';
    root.style.height = '100vh';
    document.body.appendChild(root);

    require('../src/index');
  });

  afterAll(() => {
    process.env.NODE_ENV = 'test';
    document.getElementById('root')?.remove();
  });

  it('should have mounted the app', () => {
    expect(document.getElementById('root')?.innerHTML).toMatchSnapshot();
  });
});
