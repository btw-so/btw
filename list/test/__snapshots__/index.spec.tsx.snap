// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/index should have mounted the app 1`] = `
<div>
  <div>
    root
  </div>
</div>
`;
